validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/HttpClientExternalSignerFactory.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.http.HttpClient;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.util.function.Supplier;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import org.apache.commons.lang3.tuple.Pair;
+import tech.pegasys.teku.infrastructure.crypto.SecureRandomProvider;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+import tech.pegasys.teku.validator.api.ValidatorConfig;
+
+public class HttpClientExternalSignerFactory implements Supplier<HttpClient> {
+  private final ValidatorConfig validatorConfig;
+
+  public HttpClientExternalSignerFactory(final ValidatorConfig validatorConfig) {
+    this.validatorConfig = validatorConfig;
+  }
+
+  @Override
+  public HttpClient get() {
+    final HttpClient.Builder builder = HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1);
+    if (isTLSEnabled()) {
+      builder.sslContext(
+          getSSLContext(
+              validatorConfig.getValidatorExternalSignerKeystorePasswordFilePair(),
+              validatorConfig.getValidatorExternalSignerTruststorePasswordFilePair()));
+    }
+    return builder.build();
+  }
+
+  private boolean isTLSEnabled() {
+    final String protocol = validatorConfig.getValidatorExternalSignerUrl().getProtocol();
+    return protocol != null && protocol.toLowerCase().equals("https");
```

Comment:
nit: 
```suggestion
    return protocol != null && protocol.equalsIgnoringCase("https");
```

`toLowerCase` can do surprising things in some locales and with some characters.

Reply:
done.